test/powershell/Host/TabCompletion/BugFix.Tests.ps1

```diff
@@ -35,4 +35,18 @@ Describe "Tab completion bug fix" -Tags "CI" {
         $result.CompletionMatches[1].CompletionText | Should Be "-NoClobber"
         $result.CompletionMatches[2].CompletionText | Should Be "-NoOverwrite"
     }
+    It "Issue#3416 - 'Select-Object -ExcludeProperty <tab>' should work" {
+        $cmd = "Get-Date | Select-Object -ExcludeProperty "
+        $result = TabExpansion2 -inputScript $cmd -cursorColumn $cmd.Length
+        $result.CompletionMatches.Count | Should Be 15
```

Comment:
This could be a fragile test, e.g. if a new version of .Net adds a property.

A more reliable test would look for specific results and not care about ordering.